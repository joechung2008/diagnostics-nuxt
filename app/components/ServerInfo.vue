<template>
  <UTable :data="data" :columns="columns">
    <template #name-data="{ row }">
      <span class="text-sm">{{ row.name }}</span>
    </template>
    <template #value-data="{ row }">
      <span class="text-sm">{{ row.value }}</span>
    </template>
  </UTable>
</template>

<script setup lang="ts">
import type { TableColumn } from "@nuxt/ui";
import type { KeyValuePair, ServerInfoProps } from "~/types";
import { when } from "~/utils/extensions";

const props = defineProps<ServerInfoProps>();

const data = computed(() => [
  {
    name: "Hostname",
    value: props.hostname,
  },
  ...when(props.uptime, {
    name: "Uptime",
    value: props.uptime,
  }),
  {
    name: "Server ID",
    value: props.serverId,
  },
  {
    name: "Deployment ID",
    value: props.deploymentId,
  },
  ...when(props.nodeVersions, {
    name: "Node Versions",
    value: props.nodeVersions,
  }),
  {
    name: "Extension Sync | Total Sync All Count",
    value: props.extensionSync.totalSyncAllCount,
  },
]);

const columns: TableColumn<KeyValuePair<string>>[] = [
  {
    id: "name",
    accessorKey: "name",
    header: "Name",
  },
  {
    id: "value",
    accessorKey: "value",
    header: "Value",
  },
];
</script>
