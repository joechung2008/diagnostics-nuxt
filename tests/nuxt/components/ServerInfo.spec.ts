import { mountSuspended, renderSuspended } from "@nuxt/test-utils/runtime";
import { describe, expect, it } from "vitest";
import ServerInfo from "../../../app/components/ServerInfo.vue";
import type { ServerInfoProps } from "../../../app/types";

describe("app/components/ServerInfo.vue", () => {
  it("should mount", async () => {
    const props: ServerInfoProps = {
      hostname: "test-server",
      serverId: "server-123",
      deploymentId: "deploy-456",
      extensionSync: {
        totalSyncAllCount: 10,
      },
    };

    const component = await mountSuspended(ServerInfo, {
      props,
    });
    expect(component).toBeDefined();
  });

  it("should render server info with required fields", async () => {
    const props: ServerInfoProps = {
      hostname: "production-server-01",
      serverId: "srv-2024-001",
      deploymentId: "deploy-v3.2.1",
      extensionSync: {
        totalSyncAllCount: 42,
      },
    };

    const { html } = await renderSuspended(ServerInfo, {
      props,
    });

    expect(html()).toMatchSnapshot();
  });

  it("should render server info with optional fields", async () => {
    const props: ServerInfoProps = {
      hostname: "staging-server-02",
      serverId: "srv-stage-002",
      deploymentId: "deploy-staging-v1.4",
      uptime: 123456789,
      nodeVersions: "18.17.0 LTS",
      extensionSync: {
        totalSyncAllCount: 25,
      },
    };

    const { html } = await renderSuspended(ServerInfo, {
      props,
    });

    expect(html()).toMatchSnapshot();
  });
});
