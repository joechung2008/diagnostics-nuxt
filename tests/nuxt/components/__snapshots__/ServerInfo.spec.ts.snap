// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app/components/ServerInfo.vue > should render server info with optional fields 1`] = `
"<div id="test-wrapper">
  <div class="relative overflow-auto">
    <table class="min-w-full overflow-clip">
      <!--v-if-->
      <thead class="relative">
        <tr class="data-[selected=true]:bg-elevated/50">
          <th data-pinned="false" scope="col" class="px-4 py-3.5 text-sm text-highlighted text-left rtl:text-right font-semibold [&amp;:has([role=checkbox])]:pe-0">Name</th>
          <th data-pinned="false" scope="col" class="px-4 py-3.5 text-sm text-highlighted text-left rtl:text-right font-semibold [&amp;:has([role=checkbox])]:pe-0">Value</th>
        </tr>
        <tr class="absolute z-[1] left-0 w-full h-px bg-(--ui-border-accented)"></tr>
      </thead>
      <tbody class="divide-y divide-default [&amp;>tr]:data-[selectable=true]:hover:bg-elevated/50 [&amp;>tr]:data-[selectable=true]:focus-visible:outline-primary">
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Hostname</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">staging-server-02</td>
        </tr>
        <!--v-if-->
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Uptime</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">123456789</td>
        </tr>
        <!--v-if-->
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Server ID</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">srv-stage-002</td>
        </tr>
        <!--v-if-->
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Deployment ID</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">deploy-staging-v1.4</td>
        </tr>
        <!--v-if-->
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Node Versions</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">18.17.0 LTS</td>
        </tr>
        <!--v-if-->
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Extension Sync | Total Sync All Count</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">25</td>
        </tr>
        <!--v-if-->
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>"
`;

exports[`app/components/ServerInfo.vue > should render server info with required fields 1`] = `
"<div id="test-wrapper">
  <div class="relative overflow-auto">
    <table class="min-w-full overflow-clip">
      <!--v-if-->
      <thead class="relative">
        <tr class="data-[selected=true]:bg-elevated/50">
          <th data-pinned="false" scope="col" class="px-4 py-3.5 text-sm text-highlighted text-left rtl:text-right font-semibold [&amp;:has([role=checkbox])]:pe-0">Name</th>
          <th data-pinned="false" scope="col" class="px-4 py-3.5 text-sm text-highlighted text-left rtl:text-right font-semibold [&amp;:has([role=checkbox])]:pe-0">Value</th>
        </tr>
        <tr class="absolute z-[1] left-0 w-full h-px bg-(--ui-border-accented)"></tr>
      </thead>
      <tbody class="divide-y divide-default [&amp;>tr]:data-[selectable=true]:hover:bg-elevated/50 [&amp;>tr]:data-[selectable=true]:focus-visible:outline-primary">
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Hostname</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">production-server-01</td>
        </tr>
        <!--v-if-->
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Server ID</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">srv-2024-001</td>
        </tr>
        <!--v-if-->
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Deployment ID</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">deploy-v3.2.1</td>
        </tr>
        <!--v-if-->
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Extension Sync | Total Sync All Count</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">42</td>
        </tr>
        <!--v-if-->
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>"
`;
